biorxiv/article_distances/biorxiv_article_distances.py

```diff
@@ -0,0 +1,499 @@
+    .query(f"document in {biorxiv_published.reset_index().pmcid.tolist()}")
+    .sort_values("document", ascending=True)
+    .set_index("document")
+)
+PMC_pubished.head()
+
+
+# In[10]:
+
+
+article_distances = cdist(
+    biorxiv_published
+    .loc[PMC_pubished.index.tolist()]
+    .drop(["document", "published_doi", "pmcoa"], axis=1), 
+    PMC_pubished.drop(["journal", "doi"], axis=1),
+    'euclidean'
+)
+article_distances.shape
+
+
+# In[12]:
+
+
+articles_distance_df = (
+    biorxiv_published
+    .loc[PMC_pubished.index.tolist()]
+    .reset_index()
+    [["document", "pmcid"]]
+    .assign(
+            distance=np.diag(article_distances, k=0),
+            journal=PMC_pubished.journal.tolist()
+    )
+)
+articles_distance_df.head()
+
+
+# ## biorxiv -> random paper same journal
+
+# In[13]:
+
+
+PMC_off_published = (
+    pmc_journal_mapped_df
+    .drop("doi", axis=1)
+    .query(f"document not in {biorxiv_published.reset_index().pmcid.tolist()}")
+    .query(f"journal in {articles_distance_df.journal.unique().tolist()}")
+    .groupby("journal", group_keys=False)
+    .apply(lambda x: x.sample(1, random_state=100))
+    .set_index("document")
+)
+PMC_off_published.head()
+
+
+# In[14]:
+
+
+journal_mapper = {
+    journal:col
+    for col, journal in enumerate(PMC_off_published.journal.tolist())
+}
+list(journal_mapper.items())[0:10]
+
+
+# In[17]:
+
+
+off_article_dist = cdist(
+    biorxiv_published
+    .loc[PMC_pubished.index.tolist()]
+    .drop(["document", "published_doi", "pmcoa"], axis=1), 
+    PMC_off_published.drop("journal", axis=1),
+    'euclidean'
+)
+off_article_dist.shape
+
+
+# In[18]:
+
+
+data = []
+for idx, row in tqdm.tqdm(articles_distance_df.iterrows()):
+    if row['journal'] in journal_mapper:
+        data.append(
+            {
+                "document": row['document'],
+                "pmcid":  (
+                    PMC_off_published
+                    .query(f"journal=='{row['journal']}'")
+                    .reset_index()
+                    .document
+                    .values[0]
+                ),
+                "journal": row['journal'],
+                "distance": off_article_dist[idx, journal_mapper[row['journal']]]
+            }
+        )
+
+
+# In[19]:
+
+
+final_df = (
+    articles_distance_df
+    .assign(label="pre_vs_published")
+    .append(
+        pd.DataFrame
+        .from_records(data)
+        .assign(label="pre_vs_random")
+    )
+)
+final_df.head()
+
+
+# In[20]:
+
+
+final_df = (
+    biorxiv_journal_df[["document", "doi"]]
+    .merge(final_df)
+)
+final_df.to_csv("output/article_distances.tsv", sep="\t", index=False)
+final_df.head()
+
+
+# # Distribution plot
+
+# In[21]:
+
+
+g = (
+    p9.ggplot(final_df)
+    + p9.aes(x="label",y="distance")
+    + p9.geom_boxplot()
+    + p9.coord_flip()
+    + p9.theme_seaborn()
+)
+g.save("output/biorxiv_article_distance.svg", dpi=500)
+g.save("output/biorxiv_article_distance.png", dpi=500)
+print(g)
```

Comment:
Looking at this figure, would you intrepret this as dist(preprint, some other article in the same journal ) >> dist(preprint, associated publication), which is what you'd expect. But that the distance of the preprint vs random article is close enough that you could use this to link preprint with a publication?

Are there that many articles with missing links?

Reply:
>Looking at this figure, would you interpret this as dist(preprint, some other article in the same journal ) >> dist(preprint, associated publication), which is what you'd expect. But that the distance of the preprint vs random article is close enough that you could use this to link preprint with a publication?

Correct. The central idea here is to see if there is a difference between preprints to their published version and preprints with a random articles from the same journal. Given this difference, I can devise a cutoff score that would state that this preprint and this published article should have a link because it is the same paper.

>Are there that many articles with missing links?

There are about 39k articles that are missing links, which is a bit high. I could potentially drop that number down by approximately ten percent. Ideally, if more articles were open access I could decrease it by a lot more than ten percent.